colossus/src/main/scala/colossus/service/Async.scala

```diff
@@ -14,9 +16,12 @@ trait Sender[P <: Protocol, M[_]] {
   def send(input: P#Request): M[P#Response]
 
   def disconnect()
-  
+
   def addInterceptor(interceptor: Interceptor[P]): Unit
 
+  def address(): InetSocketAddress
```

Comment:
Every sender now has an address and update method.

Reply:
I don't think it has to be solved in this PR, but it's definitely weird that `address()` returns a single address while `update` takes in sequence.  I mean it makes perfect sense in the context of what we're doing with them, but purely as methods on a trait it feels like we're trying to make `Sender` do too many things now.  We shouldn't need to have implementations that just immediately throw a "you can't do this with this subtype" exception.

Reply:
Totally agree. Going to leave that for later.